src/Ldap.php

```diff
@@ -850,11 +959,45 @@ public function bind($username = null, $password = null)
 
             $zle = new Exception\LdapException($this, $message);
         }
-        $this->disconnect();
+        $this->unbind();
 
         throw $zle;
     }
 
+    protected function shouldReconnect($resource)
+    {
+        if ($this->reconnectCount < $this->getReconnectsToAttempt()
```

Comment:
Could we reverse this ```if/else``` so that the ```else```-condition will be checked first? That would allow us to return early and skip the ```else```

Reply:
Sorry I don't understand. There's only one conditional here, e.g. it's an `else` not an `else if`, so I'm not sure what you mean by the 'else'-condition. Did you want to check `ldap_errno($resource) === -1` before `$this->reconnectCount < $this->getReconnectsToAttempt()`?

Reply:
I was thinking along these lines:
```php
protected function shouldReconnect($resource) 
{
    if ($this->reconnectCount >= $this->getReconnectsToAttempt() || ldap_errno($resource) !== -1 {
        $this->reconnectsAttempted = $this->reconnectCount;
        $this->reconnectCount = 0;
        return false;
    }
    
    $this->reconnectCount++;
    $this->reconnectSleep();
    
    try {
        $this->connect();
        $this->bind();
        $this->reconnectsAttempted = $this->reconnectCount;
        $this->reconnectCount = 0;
         return true;
    } catch (LdapException $e) {
        if ($e->getCode() !== -1) {
            return false;
        }
    }
    return $this->shouldReconnect($this->getResource());
}
```

It's just shifting code around and altering the ```if```-condition a bit but it cleans up indentation levels and also removes the not necessary ```else```s…